alphalens/performance.py

```diff
@@ -425,3 +426,123 @@ def average_cumulative_return(q_fact):
+        columns are common factor names
+        - Example:
+                                        Mkt       SMB       HML       UMD
+        2017-06-26 00:00:00+00:00    0.0004   -0.0008    0.0069   -0.0057
+        2017-06-27 00:00:00+00:00   -0.0084   -0.0043    0.0130   -0.0030
+        2017-06-28 00:00:00+00:00    0.0102    0.0077    0.0021    0.0091
+    hierarchy : OrderedDict
+        Hierarchy of common factors, for use in plotting the bar chart.
+        If None, defaults to a hierarchy in which each column of risk_factors
+        forms a 'level' of the hierarchy.
+        - Example:
+        OrderedDict([('Market', ['Mkt']),
+                     ('Style', ['SMB', 'HML', 'UMD'])])
+    compute_t_stats : bool
+        If True, returns t-statistics for the betas. Defaults to False
+    compute_vifs : bool
+        If True, computes and returns variance inflation factors for the
+        common factors. Defaults to False
+
+    Returns
+    -------
+    returns_decomposition : pd.DataFrame
+        Decomposed returns
+
+    betas : pd.DataFrame
+        Betas to the common factors
+
+    vifs : pd.Series
+        Variance inflation factors for the common factors
+        - For more information, see
+        https://en.wikipedia.org/wiki/Variance_inflation_factor
+    '''
+    if hierarchy is None:
+        hierarchy = risk_factors.columns.values
+
+    # construct a cumulative hierarchy
+    cum_tiers = []
+    for i in range(len(hierarchy.values())):
+        cum_tiers.append([item for a in range(0, i+1)
+                          for item in hierarchy.values()[a]])
+    cum_hierarchy = OrderedDict()
+    cum_hierarchy.update((hierarchy.keys()[i], cum_tiers[i])
+                         for i in range(len(hierarchy)))
+
+    # ensure that risk_factors and algo_rets_over_rf have the same index
+    idx = algo_returns.index.intersection(risk_factors.index)
+    risk_factors = risk_factors.loc[idx]
+    algo_returns = algo_returns.loc[idx]
+
+    idx = np.append(['Alpha'], risk_factors.columns.values)
+    betas = pd.DataFrame(index=idx, columns=hierarchy.keys())
+    returns_decomposition = pd.DataFrame(index=idx, columns=hierarchy.keys())
+
+    if compute_t_stats:
+        t_stats = pd.DataFrame(index=idx, columns=hierarchy.keys())
+
+    for tier, factors in cum_hierarchy.iteritems():
+        model_factors = add_constant(risk_factors.loc[:, factors]) \
+            .rename(columns={'const': 'Alpha'})
+        model = OLS(algo_returns, model_factors).fit()
+
+        betas.loc[:, tier] = model.params
+        betas.loc['Alpha', tier] *= 252
+
+        if compute_t_stats:
+            t_stats.loc[:, tier] = model.params / model.HC0_se
+
+        returns_decomposition.loc[:, tier] = model.params.drop('Alpha') \
+            * risk_factors.mean() * 252
+        returns_decomposition.loc['Alpha', tier] = betas.loc['Alpha', tier]
+
+    if compute_t_stats and compute_vifs:
```

Comment:
I like the idea of displaying a warning about the vifs. I think we may as well return all 4 vars while we're at it.